src/fieldsArray.js

```diff
@@ -6,98 +6,65 @@ const debug = require("debug")("passkit:fields");
  * @see https://apple.co/2wkUBdh
  */
 
-class FieldsArray extends Set {
-  constructor(fields) {
-    super();
-
-    /**
-     * Pass fields must be unique (for key) in its scope.
-     * Therefore we use a Set to keep them tracked.
-     */
-
-    this.fieldsKeys = new Set();
-    fields.forEach(a => this[a] = new ItemsArray(this));
-  }
-
-  addFieldKey(key) {
-
-    if (this.fieldsKeys.has(key)) {
-      debug(`Field with key "${key}" discarded: fields must be unique in pass scope.`);
-    } else {
-      this.fieldsKeys.add(key);
-    }
-  }
-
-  deleteFieldKey(key) {
-    this.fieldsKeys.delete(key);
-  }
-
-  emptyUnique() {
-    this.fieldsKeys.clear();
-  }
+const poolSymbol = Symbol("pool");
+
+class FieldsArray extends Array {
+	
+	constructor(pool,...args) {
+    	super(...args);
+    	this[poolSymbol] = pool;
+	}
+
+	/**
+	 * Like `Array.prototype.push` but will alter
+	 * also uniqueKeys set.
+	 */
+
+	push(...fieldsData) {
+		const validFields = fieldsData.reduce((acc, current) => {
+			if (!(typeof current === "object") || !schema.isValid(current, "field")) {
+				return acc;
+			}
+
+			if (acc.some(e => e.key === current.key) || this[poolSymbol].has(current.key)) {
```

Comment:
@ianbale I found right now that this logic is completely broken: first, the name of debug is `fieldsDebug`. Second, the field must be pushed into the accumulator only it the debug condition is false... 😢